scripts/upgrade/sabnzbd.sh

```diff
@@ -1,40 +1,63 @@
 #!/bin/bash
-user=$(cut -d: -f1 < /root/.master.info)
+user=$(cut -d: -f1 </root/.master.info)
 
 if [[ ! -f /install/.sabnzbd.lock ]]; then
     echo_error "SABnzbd not detected. Exiting!"
     exit 1
 fi
 
+. /etc/swizzin/sources/functions/pyenv
 localversion=$(/opt/.venv/sabnzbd/bin/python /opt/sabnzbd/SABnzbd.py --version | grep -m1 SABnzbd | cut -d- -f2)
 #latest=$(curl -s https://sabnzbd.org/downloads | grep -m1 Linux | grep download-link-src | grep -oP "href=\"\K[^\"]+")
 latest=$(curl -sL https://api.github.com/repos/sabnzbd/sabnzbd/releases/latest | grep -Po '(?<="browser_download_url":).*?[^\\].tar.gz"' | sed 's/"//g')
 latestversion=$(echo $latest | awk -F "/" '{print $NF}' | cut -d- -f2)
+pyvenv_version=$(/opt/.venv/sabnzbd/bin/python --version | awk '{print $2}')
 
-LIST='par2 p7zip-full python3-dev python3-setuptools python3-pip python3-venv libffi-dev libssl-dev libglib2.0-dev libdbus-1-dev'
+if dpkg --compare-versions ${pyvenv_version} lt 3.6.0 && dpkg --compare-versions ${latestversion} ge 3.2.0; then
+    LIST='par2 p7zip-full libffi-dev libssl-dev libglib2.0-dev libdbus-1-dev'
+    PYENV_REBUILD=True
+elif [[ -f /opt/.venv/sabnzbd/bin/python2 ]]; then
+    LIST='par2 p7zip-full python3-dev python3-setuptools python3-pip python3-venv libffi-dev libssl-dev libglib2.0-dev libdbus-1-dev'
+    PYENV_REBUILD=True
+else
+    LIST='par2 p7zip-full python3-dev python3-setuptools python3-pip python3-venv libffi-dev libssl-dev libglib2.0-dev libdbus-1-dev'
+fi
 apt_install $LIST
 
 if dpkg --compare-versions ${localversion} lt ${latestversion}; then
-    # echo "Upgrading SABnzbd ... "
+    if [[ $PYENV_REBUILD == True ]]; then
+        echo_progress_start "Upgrading SABnzbd python virtual environment"
+        rm -rf /opt/.venv/sabnzbd
+        systempy3_ver=$(get_candidate_version python3)
+        if dpkg --compare-versions ${systempy3_ver} lt 3.6.0; then
+            pyenv_install
+            pyenv_install_version 3.7.7
+            pyenv_create_venv 3.7.7 /opt/.venv/sabnzbd
+            chown -R ${user}: /opt/.venv/sabnzbd
+        else
+            python3_venv ${user} sabnzbd
+        fi
+        if grep -q python2 /etc/systemd/system/sabnzbd.service; then
+            sed -i 's/python2/python/g' /etc/systemd/system/sabnzbd.service
+            systemctl daemon-reload
+        fi
+
+        echo_progress_done
+    fi
     cd /opt/
```

Comment:
Any chance we get rid of the `cd`s?

Reply:
What harm is the `cd` causing here? I'd rather the tarball get downloaded into opt than `cwd`

Reply:
the `wget` can still do something like `-O /tmp/archive.tar.gz`. Downloading to `cwd` is a sin, and `cd` being successful should not be taken for granted

Reply:
> cd being successful should not be taken for granted
True but we also create the directory in the directly preceding command. Shellcheck is not god and it's suggestions have led me astray from time to time 😉

Reply:
I'm suggesting this
```bash
wget $link -O /tmp/archive.tar.gz
tar xzf /tmp/archive.targz -C /opt/sbanznznzd --strip
rm /tmp/archive.tar.gz
```
I'm not even referring to shellcheck here, `cd`ing and manipulating `cwd` is just not a great idea imo